configure.ac

```diff
@@ -98,23 +97,45 @@ fi
 AM_CONDITIONAL(SYSTEMD, test -n "${with_systemdunitdir}")
 
 AC_ARG_WITH([dbuspolicydir],
-        AS_HELP_STRING([--with-dbuspolicydir=DIR], [D-Bus policy directory]),
-        [],
-        [with_dbuspolicydir="$($PKG_CONFIG --variable=datadir dbus-1)/dbus-1/system.d"])
+	    AS_HELP_STRING([--with-dbuspolicydir=DIR], [D-Bus policy directory]),
+	    [autodetect_dbuspolicydir=false],
+	    [autodetect_dbuspolicydir=true])
+
+AC_ARG_WITH([dbussystemservicedir],
+	    AS_HELP_STRING([--with-dbussystemservicedir=DIR], [D-Bus system service directory]),
+	    [autodetect_dbussystemservicedir=false],
+	    [autodetect_dbussystemservicedir=true])
+
+AC_ARG_WITH([dbusinterfacesdir],
+	    AS_HELP_STRING([--with-dbusinterfacesdir=DIR], [D-Bus interfaces directory]),
+	    [autodetect_dbusinterfacesdir=false],
+	    [autodetect_dbusinterfacesdir=true])
+
+AS_IF([$autodetect_dbuspolicydir || $autodetect_dbussystemservicedir || $autodetect_dbusinterfacesdir], [
+       PKG_CHECK_MODULES([DBUS], [dbus-1])
+
+       AS_IF([$autodetect_dbuspolicydir], [
+	      AC_MSG_CHECKING([dbus policy dir])
+	      with_dbuspolicydir="$($PKG_CONFIG --variable=datadir dbus-1)/dbus-1/system.d"
+	      AC_MSG_RESULT([$with_dbuspolicydir])])
```

Comment:
The `pkg-config` man-page suggests to use
`PKG_CHECK_VAR(VARIABLE, MODULE, CONFIG-VARIABLE, [ACTION-IF-FOUND], [ACTION-IF-NOT-FOUND])`

Reply:
This is IMHO a separate issue as the pkg-config call is only moved. So I'd address this in a separate change.

Reply:
pushed another commit on top of the previous one to convert to `PKG_CHECK_VAR`.